timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -229,9 +231,15 @@ def _validate_multi_index_data_frame(cls, data: pd.DataFrame):
         if not isinstance(data.index, pd.MultiIndex):
             raise ValueError(f"data must have pd.MultiIndex, got {type(data.index)}")
         if not data.index.dtypes.array[1] == "datetime64[ns]":
-            raise ValueError(f"for {TIMESTAMP}, the only pandas dtype allowed is ‘datetime64[ns]’.")
+            raise ValueError(f"for {TIMESTAMP}, the only pandas dtype allowed is `datetime64[ns]`.")
         if not data.index.names == (f"{ITEMID}", f"{TIMESTAMP}"):
             raise ValueError(f"data must have index names as ('{ITEMID}', '{TIMESTAMP}'), got {data.index.names}")
+        item_id_index = data.index.get_level_values(level=ITEMID)
+        # workaround for pd.api.types.is_string_dtype issue https://github.com/pandas-dev/pandas/issues/15585
+        item_id_is_string = (item_id_index == item_id_index.astype(str)).all()
+        item_id_is_int = pd.api.types.is_integer_dtype(item_id_index)
+        if not (item_id_is_string or item_id_is_int):
+            raise ValueError(f"all entries in index `{ITEMID}` must be of integer or string dtype")
```

Comment:
We now always ensure that the `ITEMID` columns of the index either contains all integers or all strings. This will avoid potential problems, such as collision of two time series with `item_id`s `0` (int) and `"0"` (str).